### PR TITLE
Dubbo-samples-stub Refactor to springboot style

### PR DESCRIPTION
**Pull request by mufiye**
Change the 2-advanced dubbo-samples-stub code "spring xml style" to "spring boot style". Wish to be reviewed.

### CODE REVIEW

**Comment by mufiye** [comment]
I have checked the workflow results, and I find that many other modules fail but I think I don't change them. For my dubbo-samples-stub module, I run its integration test locally with 3.2.0-beta.4 successfully. But if run with 3.2.0-beta.5, the below error will appear.
```
java.lang.IllegalStateException: Extension instance (name: observationreceiver, class: interface org.apache.dubbo.rpc.Filter) couldn't be instantiated: null
	at org.apache.dubbo.common.extension.ExtensionLoader.createExtension(ExtensionLoader.java:826)
	at org.apache.dubbo.common.extension.ExtensionLoader.getExtension(ExtensionLoader.java:575)
	at org.apache.dubbo.common.extension.ExtensionLoader.getExtension(ExtensionLoader.java:549)
	at org.apache.dubbo.common.extension.ExtensionLoader.lambda$getActivateExtension$4(ExtensionLoader.java:398)
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:721)
	at java.base/java.util.Collections$SynchronizedMap.forEach(Collections.java:2735)
	at org.apache.dubbo.common.extension.ExtensionLoader.getActivateExtension(ExtensionLoader.java:393)
	at org.apache.dubbo.common.extension.ExtensionLoader.getActivateExtension(ExtensionLoader.java:331)
	at org.apache.dubbo.rpc.cluster.filter.DefaultFilterChainBuilder.buildInvokerChain(DefaultFilterChainBuilder.java:50)
	at org.apache.dubbo.rpc.cluster.filter.ProtocolFilterWrapper.export(ProtocolFilterWrapper.java:61)
	at org.apache.dubbo.rpc.protocol.ProtocolSerializationWrapper.export(ProtocolSerializationWrapper.java:47)
	at org.apache.dubbo.rpc.Protocol$Adaptive.export(Protocol$Adaptive.java)
	at org.apache.dubbo.config.ServiceConfig.doExportUrl(ServiceConfig.java:738)
	at org.apache.dubbo.config.ServiceConfig.exportLocal(ServiceConfig.java:754)
	at org.apache.dubbo.config.ServiceConfig.exportUrl(ServiceConfig.java:640)
	at org.apache.dubbo.config.ServiceConfig.doExportUrlsFor1Protocol(ServiceConfig.java:450)
	at org.apache.dubbo.config.ServiceConfig.doExportUrls(ServiceConfig.java:432)
	at org.apache.dubbo.config.ServiceConfig.doExport(ServiceConfig.java:394)
	at org.apache.dubbo.config.ServiceConfig.export(ServiceConfig.java:246)
	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.exportServiceInternal(DefaultModuleDeployer.java:350)
	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.exportServices(DefaultModuleDeployer.java:322)
	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.startSync(DefaultModuleDeployer.java:158)
	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.start(DefaultModuleDeployer.java:139)
	at org.apache.dubbo.config.spring.context.DubboDeployApplicationListener.onContextRefreshedEvent(DubboDeployApplicationListener.java:113)
	at org.apache.dubbo.config.spring.context.DubboDeployApplicationListener.onApplicationEvent(DubboDeployApplicationListener.java:102)
	at org.apache.dubbo.config.spring.context.DubboDeployApplicationListener.onApplicationEvent(DubboDeployApplicationListener.java:47)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:176)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:169)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:143)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:413)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:370)
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:935)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:587)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:730)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:432)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:308)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1302)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1291)
	at org.apache.dubbo.samples.stub.provider.StubProvider.main(StubProvider.java:33)
```

**Comment by chickenlj** [comment]
> I have checked the workflow results, and I find that many other modules fail but I think I don't change them. For my dubbo-samples-stub module, I run its integration test locally with 3.2.0-beta.4 successfully. But if run with 3.2.0-beta.5, the below error will appear.
> 
> ```java
> java.lang.IllegalStateException: Extension instance (name: observationreceiver, class: interface org.apache.dubbo.rpc.Filter) couldn't be instantiated: null
> 	at org.apache.dubbo.common.extension.ExtensionLoader.createExtension(ExtensionLoader.java:826)
> 	at org.apache.dubbo.common.extension.ExtensionLoader.getExtension(ExtensionLoader.java:575)
> 	at org.apache.dubbo.common.extension.ExtensionLoader.getExtension(ExtensionLoader.java:549)
> 	at org.apache.dubbo.common.extension.ExtensionLoader.lambda$getActivateExtension$4(ExtensionLoader.java:398)
> 	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:721)
> 	at java.base/java.util.Collections$SynchronizedMap.forEach(Collections.java:2735)
> 	at org.apache.dubbo.common.extension.ExtensionLoader.getActivateExtension(ExtensionLoader.java:393)
> 	at org.apache.dubbo.common.extension.ExtensionLoader.getActivateExtension(ExtensionLoader.java:331)
> 	at org.apache.dubbo.rpc.cluster.filter.DefaultFilterChainBuilder.buildInvokerChain(DefaultFilterChainBuilder.java:50)
> 	at org.apache.dubbo.rpc.cluster.filter.ProtocolFilterWrapper.export(ProtocolFilterWrapper.java:61)
> 	at org.apache.dubbo.rpc.protocol.ProtocolSerializationWrapper.export(ProtocolSerializationWrapper.java:47)
> 	at org.apache.dubbo.rpc.Protocol$Adaptive.export(Protocol$Adaptive.java)
> 	at org.apache.dubbo.config.ServiceConfig.doExportUrl(ServiceConfig.java:738)
> 	at org.apache.dubbo.config.ServiceConfig.exportLocal(ServiceConfig.java:754)
> 	at org.apache.dubbo.config.ServiceConfig.exportUrl(ServiceConfig.java:640)
> 	at org.apache.dubbo.config.ServiceConfig.doExportUrlsFor1Protocol(ServiceConfig.java:450)
> 	at org.apache.dubbo.config.ServiceConfig.doExportUrls(ServiceConfig.java:432)
> 	at org.apache.dubbo.config.ServiceConfig.doExport(ServiceConfig.java:394)
> 	at org.apache.dubbo.config.ServiceConfig.export(ServiceConfig.java:246)
> 	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.exportServiceInternal(DefaultModuleDeployer.java:350)
> 	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.exportServices(DefaultModuleDeployer.java:322)
> 	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.startSync(DefaultModuleDeployer.java:158)
> 	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.start(DefaultModuleDeployer.java:139)
> 	at org.apache.dubbo.config.spring.context.DubboDeployApplicationListener.onContextRefreshedEvent(DubboDeployApplicationListener.java:113)
> 	at org.apache.dubbo.config.spring.context.DubboDeployApplicationListener.onApplicationEvent(DubboDeployApplicationListener.java:102)
> 	at org.apache.dubbo.config.spring.context.DubboDeployApplicationListener.onApplicationEvent(DubboDeployApplicationListener.java:47)
> 	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:176)
> 	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:169)
> 	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:143)
> 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:413)
> 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:370)
> 	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:935)
> 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:587)
> 	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:730)
> 	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:432)
> 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:308)
> 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1302)
> 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1291)
> 	at org.apache.dubbo.samples.stub.provider.StubProvider.main(StubProvider.java:33)
> ```

Yes, it's caused by other changes in apache/dubbo repo.

File: 2-advanced/dubbo-samples-stub/dubbo-samples-stub-provider/src/main/java/org/apache/dubbo/samples/stub/provider/StubProvider.java [hunk 1]
```diff
@@ -17,19 +17,20 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 import java.util.concurrent.CountDownLatch;
 
+@SpringBootApplication
+@EnableDubbo
 public class StubProvider {
-
     public static void main(String[] args) throws Exception {
-        new EmbeddedZooKeeper(2181, false).start();
```
Comment (chickenlj) [inline 2-advanced/dubbo-samples-stub/dubbo-samples-stub-provider/src/main/java/org/apache/dubbo/samples/stub/provider/StubProvider.java:29]: Keeping an embedded zookeeper would make it easier for users to use.

**Review by chickenlj — APPROVED** [review]
LGTM

**Pull request merged**